docs/go-configuration.md

```diff
@@ -4,27 +4,599 @@ title: Scope Go Agent configuration
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
 
 ```sh
 SCOPE_TESTING_MODE=false
 ```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithTestingModeEnabled()))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithTestingModeEnabled()))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Enabling test retries on failure
+
+The Scope agent has support for retrying failed tests. By default, no retries are performed.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_TESTING_FAIL_RETRIES=3
+```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithRetriesOnFail(3)))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithRetriesOnFail(3)))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Handling test panic as a failure
+
+By default, the standard testing framework exits the test process on panic. This behavior can be changed to handle panics as test failures, allowing the complete execution of the test suite. 
+
+Also, this option can be combined with test retries, to retry tests that panic.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_TESTING_PANIC_AS_FAIL=true
+```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithHandlePanicAsFail()))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithHandlePanicAsFail()))
```

Comment:
This doesn't seem like runtime instrumentation

Reply:
ok